onchain/rollups/test/foundry/consensus/quorum/QuorumFactory.t.sol

```diff
@@ -0,0 +1,121 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+/// @title Quorum Factory Test
+pragma solidity ^0.8.8;
+
+import {Test} from "forge-std/Test.sol";
+import {QuorumFactory} from "contracts/consensus/quorum/QuorumFactory.sol";
+import {Quorum} from "contracts/consensus/quorum/Quorum.sol";
+import {IHistory} from "contracts/history/IHistory.sol";
+import {Vm} from "forge-std/Vm.sol";
+
+contract QuorumFactoryTest is Test {
+    QuorumFactory factory;
+    address[] validators;
+    uint256[] shares;
+    uint256 numValidator;
+
+    event QuorumCreated(Quorum quorum, IHistory history);
+
+    function setUp() public {
+        factory = new QuorumFactory();
+    }
+
+    function testNewQuorum(uint16 _numValidator, IHistory _history) public {
+        generateAddrAndShares(_numValidator);
+
+        vm.recordLogs();
+
+        Quorum quorum = factory.newQuorum(validators, shares, _history);
+
+        testNewQuorumAux(validators, shares, _history, quorum);
+    }
+
+    function generateAddrAndShares(uint256 _numValidator) internal {
+        // assume there are less than 1000 validators so the test wont run forever
+        vm.assume(_numValidator > 0 && _numValidator < 1000);
+        numValidator = _numValidator;
+
+        validators = new address[](numValidator);
+        shares = new uint256[](numValidator);
+        for (uint256 i; i < numValidator; ++i) {
+            validators[i] = vm.addr(i + 1); // assume there is no address collision
+            shares[i] = i + 1;
```

Comment:
Maybe we could hash `i`, cast it into a `uint256`, and force it to be contained in the $(1,(2^{256}-1)/n)$ interval with `bound`, where $n$ is the number of validators. This way, we guarantee that...

$$
(\forall i \in [1,n], shares_i > 0) \wedge \sum_{i=1}^n shares_i < 2^{256}
$$

... which is required by `PaymentSplitter` [(1)](https://github.com/binodnp/openzeppelin-solidity/blob/ffc4bcbba398b61d19ede03e4fdf673ddbce1083/contracts/payment/PaymentSplitter.sol#L108) [(2)](https://github.com/binodnp/openzeppelin-solidity/blob/ffc4bcbba398b61d19ede03e4fdf673ddbce1083/contracts/payment/PaymentSplitter.sol#L113).

Reply:
To sum an arithmetic sequence to $2^{256}$, we would need to have about $2^{128}$ validators. I don't think the quorum would get that large

Reply:
> To sum an arithmetic sequence to 2256, we would need to have about 2128 validators. I don't think the quorum would get that large

Yes, but if you hash each element of the arithmetic sequence, you might get some large numbers that can add up to more than $2^{256}$. Either way, I might be overcomplicating things and using the arithmetic sequence like you are already doing.